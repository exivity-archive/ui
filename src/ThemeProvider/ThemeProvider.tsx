import React from 'react'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
// @ts-ignore
import WebfontLoader from '@dr-kobros/react-webfont-loader'

import defaultTheme, { Theme } from '../defaultTheme/theme'

interface ThemeProviderProps {
  theme?: Theme
  onFontStatus?: (status: string) => void
  children?: any
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  theme = defaultTheme,
  onFontStatus,
  children
}) => (
  <StyledThemeProvider theme={theme}>
    <WebfontLoader config={theme.fonts} onStatus={onFontStatus && onFontStatus}>
      {children}
    </WebfontLoader>
  </StyledThemeProvider>
)

export default ThemeProvider
