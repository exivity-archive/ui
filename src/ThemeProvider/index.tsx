import React from 'react'
import { ThemeProvider } from 'styled-components'
// @ts-ignore
import WebfontLoader from '@dr-kobros/react-webfont-loader'

import mainTheme, { Theme } from '../defaultTheme/theme'

interface ThemeAndFontProviderProps {
  theme?: Theme
  onFontStatus?: (status: string) => void
  children: any
}

export const ThemeAndFontProvider: React.FC<ThemeAndFontProviderProps> = ({
    theme = mainTheme,
    onFontStatus,
    children
}) => (
    <ThemeProvider theme={theme}>
        <WebfontLoader config={theme.fonts} onStatus={onFontStatus && onFontStatus}>
            { children }
        </WebfontLoader>
    </ThemeProvider>
)

export default ThemeAndFontProvider
