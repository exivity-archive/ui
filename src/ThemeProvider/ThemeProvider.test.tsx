import renderer from 'react-test-renderer'
import * as React from 'react'

import ThemeProvider from '.'

test('renders basic ThemeProvider', () => {
  const provider = renderer.create(<ThemeProvider />)
  expect(provider.toJSON()).toMatchSnapshot()
})
