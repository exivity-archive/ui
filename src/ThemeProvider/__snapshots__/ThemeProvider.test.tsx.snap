// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders basic ThemeProvider 1`] = `null`;
